index.js

```diff
@@ -149,6 +150,90 @@ class GithubScm extends Scm {
     }
 
     /**
+     * Adds the Screwdriver webhook to the Github repository
+     * @method _addWebhook
+     * @param  {Object}    config            Config object
+     * @param  {String}    config.scmUri     The SCM URI to add the webhook to
+     * @param  {String}    config.token      Service token to authenticate with Github
+     * @param  {String}    config.webhookUrl The URL to use for the webhook notifications
+     * @return {Promise}                     Resolve means operation completed without failure.
+     */
+    _addWebhook(config) {
+        const searchForHook = (scmInfo, url, page) => {
+            const getHooks = this.breaker.runCommand({
+                action: 'getHooks',
+                token: config.token,
+                params: {
+                    owner: scmInfo.owner,
+                    repo: scmInfo.repo,
+                    page,
+                    per_page: WEBHOOK_PAGE_SIZE
+                }
+            });
+            const filterForHook = getHooks.then((hooks) => {
+                let found = null;
+
+                hooks.some((hook) => {
+                    const currentHook = hoek.reach(hook, 'config.url');
+
+                    if (currentHook === url) {
+                        found = hook;
+                    }
+
+                    return found;
+                });
+
+                return found;
+            });
+
+            return Promise.all([
+                getHooks,
+                filterForHook
+            ]).then(([hooks, screwdriverHook]) => {
+                if (!screwdriverHook && hooks.length === WEBHOOK_PAGE_SIZE) {
+                    return searchForHook(scmInfo, url, page + 1);
+                }
+
+                return screwdriverHook;
+            });
+        };
+        const lookupScm = this.lookupScmUri({
+            scmUri: config.scmUri,
+            token: config.token
+        });
+
+        return Promise.all([
+            lookupScm,
+            lookupScm.then(scmInfo => searchForHook(scmInfo, config.webhookUrl, 1))
```

Comment:
Will `lookupScm.then` wait for `lookupScm` to finish since they are both in Promise.all? Why not just call lookupScm and then call searchForHook?